src/render/glyph_atlas.js

```diff
@@ -1,12 +1,14 @@
 // @flow
 
+import {SDF_SCALE} from '../render/glyph_manager';
 import {AlphaImage} from '../util/image';
 import {register} from '../util/web_worker_transfer';
 import potpack from 'potpack';
 
 import type {GlyphMetrics, StyleGlyph} from '../style/style_glyph';
 
-const padding = 1;
+const glyphPadding = 1;
+const localGlyphPadding = glyphPadding * SDF_SCALE;
```

Comment:
So, I've looked up the original comment that explained this 1-px padding:

```js
// When copied into the atlas texture, image data is padded by one pixel on each side. Icon
// images are padded with fully transparent pixels, while pattern images are padded with a
// copy of the image data wrapped from the opposite side. In both cases, this ensures the
// correct behavior of GL_LINEAR texture sampling mode.
```

It seems that since this is for sampling, we don't need to scale it with SDF_SCALE, and just use 1px uniformly for all glyphs. Or not?

Reply:
Yeah, the only reason for scaling the padding here is to keep the entirety of the texture scaled at the same multiple so that in `quads.js` we can just scale it back down by SDF_SCALE to get the same layout/positioning/sizing as if the scale were 1. If we left the padding at 1px here, then at quad generation time we'd have to do a tiny bit of math for each glyph to figure out how to scale it down by a factor slightly less than SDF_SCALE. It seemed to me like a relatively small cost either way and scaling the padding was simpler. I can put in comments to that effect.

I haven't thought this through, but I guess since the padding just has to be in the texture (for sampling purposes) but isn't meant to be rendered itself, it probably doesn't need to be in the "positions" returned from the glyph atlas (and passed through shaping and quad generation to the shaders)? If we made a change like that (and made sure it was a no-op for server-side glyphs), then the SDF_SCALE would be irrelevant. Is that something like what you had in mind?